packages/agents/src/mcp.ts

```diff
@@ -170,20 +393,27 @@ export abstract class McpAgent<
               { status: 400 }
             );
           }
+
+          // Get the Durable Object
           const object = namespace.get(namespace.idFromString(sessionId));
-          const response = await object.onMCPMessage(request);
 
-          // Convert headers to a plain object
-          const headerObj: Record<string, string> = {};
-          response.headers.forEach((value, key) => {
-            headerObj[key] = value;
+          // Forward the request to the Durable Object
+          const response = await object.onMCPMessage(sessionId, request);
+
+          // Add CORS headers
+          const headers = new Headers();
+          response.headers.forEach?.((value, key) => {
+            headers.set(key, value);
           });
-          headerObj["Access-Control-Allow-Origin"] = corsOptions?.origin || "*";
+          headers.set(
+            "Access-Control-Allow-Origin",
+            corsOptions?.origin || "*"
+          );
 
-          return new Response(response.body as unknown as BodyInit, {
+          return new Response(await response.text(), {
```

Comment:
Probably should just pass `response.body` here, since that'll stream. Not that it'll matter for this, just a habit to get into I guess